contracts/starknet/lib/hash_array.cairo

```diff
@@ -1,13 +1,20 @@
 from starkware.cairo.common.cairo_builtins import HashBuiltin
 from starkware.cairo.common.hash_state import hash_init, hash_update
+from starkware.cairo.common.alloc import alloc
 
 # Internal utility function to hash felt arrays.
 # Dev note: starkware.py and starknet.js methods for hashing an array append the length of the array to the end before hashing.
 # This is why we replicate that here.
 func hash_array{pedersen_ptr : HashBuiltin*}(array_len : felt, array : felt*) -> (hash : felt):
-    # Appending the length of the array to itself as the offchain version of the hash works this way
-    assert array[array_len] = array_len
+    # Hash the array
     let (hash_state_ptr) = hash_init()
-    let (hash_state_ptr) = hash_update{hash_ptr=pedersen_ptr}(hash_state_ptr, array, array_len + 1)
+    let (hash_state_ptr) = hash_update{hash_ptr=pedersen_ptr}(hash_state_ptr, array, array_len)
+
+    # Append the length of the array to itself as the offchain version of the hash works this way
+    let (suffix : felt*) = alloc()
+    assert suffix[0] = array_len
```

Comment:
Rather than allocating a new array, might be cheaper to access the pointer directly. Like:

```
    let (__fp__, _) = get_fp_and_pc()

    let (hash_state_ptr) = hash_update{hash_ptr=pedersen_ptr}(hash_state_ptr, &array_len, 1)
```

Reply:
The problem I had was with the `&array_len`. Does `get_fp_and_pc()` fix that?

Reply:
yep, it allows you to access the pointer

Reply:
I'm surprised this works! By updating `fp` I suddenly can take the address of `array_len`. Wild!
https://github.com/snapshot-labs/sx-core/pull/269/commits/94d07e0d867327656f6c96ddd6c72edf7c3b937d